sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -393,6 +393,21 @@ case class PreprocessTableInsertion(conf: SQLConf) extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * SPARK-28443: Spark sql add exception when create field type NullType
+ */
+object CreateTableCheck extends Rule[LogicalPlan] {
+  override def apply(plan: LogicalPlan): LogicalPlan = {
+    plan match {
+      case ct: CreateTable if ct.tableDesc.schema.exists { f =>
```

Comment:
can we move it to the rule `PreWriteCheck`?

Reply:
I consider that it's a little different between create table(DDL) and insert data (DML).  Maybe split in two rules ?

Reply:
yea we can add a `DDLCheck` rule. Please follow `PreWriteCheck` to make it a pure-checking rule.

Also don't forget to handle DS v2 CREATE TABLE as well, which has a different logical plan.
- `CreateV2Table`
- `CreateTableAsSelect`
- `ReplaceTable`
- `ReplaceTableAsSelect`

also cc @rdblue

Reply:
ok, I will check this later